### PR TITLE
Add formatString as replacement for gmx::formatString

### PR DESCRIPTION
**Pull request by ejjordan**
Also removed 2 unused includes.

### CODE REVIEW

File: src/gromacs/nblib/interactions.cpp [hunk 1]
```diff
@@ -119,9 +119,9 @@ void ParticleTypesInteractions::add(const ParticleTypeName& particleTypeName1,
         if (std::get<0>(insertLocation.first->second) != c6
             || std::get<1>(insertLocation.first->second) != c12)
         {
-            std::string message = gmx::formatString(
+            std::string message = formatString(
                     "Attempting to add nonbonded interaction parameters between the particle types "
-                    "%s %s twice",
+                    "{} {} twice",
                     particleTypeName1.c_str(), particleTypeName2.c_str());
```
Comment (sekelle) [inline src/gromacs/nblib/interactions.cpp]: `.c_str()` is not needed anymore

File: src/gromacs/nblib/interactions.cpp [hunk 2]
```diff
@@ -174,9 +174,8 @@ NonBondedInteractionMap ParticleTypesInteractions::generateTable()
             auto interactionKey = std::make_tuple(particleTypeName1, particleTypeName2);
             if (nonbondedParameters_.count(interactionKey) == 0)
             {
-                std::string message =
-                        gmx::formatString("Missing interaction between %s %s",
-                                          particleTypeName1.c_str(), particleTypeName2.c_str());
```
Comment (sekelle) [inline src/gromacs/nblib/interactions.cpp:179]: same

File: src/gromacs/nblib/topology.cpp [hunk 3]
```diff
@@ -248,8 +247,8 @@ Topology TopologyBuilder::buildTopology()
             if (topology_.nonBondedInteractionMap_.count(interactionKey) == 0)
             {
                 std::string message =
-                        gmx::formatString("Missing nonbonded interaction parameters for pair %s %s",
-                                          particleType1.first.c_str(), particleType2.first.c_str());
```
Comment (sekelle) [inline src/gromacs/nblib/topology.cpp:252]: same

File: src/gromacs/nblib/util.h [hunk 4]
```diff
@@ -231,6 +236,20 @@ auto binary_fold(Op&& op, First&& first, Ts&&... args)
     return op(first, binary_fold(op, args...));
 }
 
+template<class... Args>
+std::string formatString(std::string fmt, Args... args)
+{
+    std::ostringstream os;
+    std::string        delimiter = "{}";
+
+    __attribute__((unused)) std::initializer_list<int> unused{
```
Comment (sekelle) [inline src/gromacs/nblib/util.h]: Nice. Is it portable though?

**Review by sekelle — CHANGES_REQUESTED** [review]

File: src/gromacs/nblib/util.h (hunk 4 above)
Comment (ejjordan) [reply src/gromacs/nblib/util.h]: Probably only works for gcc and clang https://nelkinda.com/blog/suppress-warnings-in-gcc-and-clang/

**Review by victorusu — APPROVED** [review]
`lgtm`

**Review by kanduri — APPROVED** [review]
LGTM

**Review by sekelle — APPROVED** [review]
lgtm

**Pull request merged**